### PR TITLE
Add docs for i18n in paper

### PR DESCRIPTION
**Pull request by Machine-Maker**
part of https://github.com/PaperMC/docs/issues/190

### CODE REVIEW

File: docs/paper/dev/api/component-api/i18n.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+slug: /dev/component-api/i18n
+---
+
+# Internationalization
+
+Generally it's a good idea to support translations in your plugin, especially if you want to
+appeal to the largest userbase. Paper and Adventure make this simply by adding a server-side
```
Comment (kennytv) [inline docs/paper/dev/api/component-api/i18n.md:8]:
```suggestion
appeal to the largest userbase. Paper and Adventure make this simple by adding a server-side
```

File: docs/paper/dev/api/component-api/i18n.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+the [Translator](https://jd.advntr.dev/api/latest/net/kyori/adventure/translation/Translator.html) interface yourself.
+
+## Where translations work
+
+Vanilla minecraft handles translations on the client by using the language files bundled with the client or provided via resource packs. If you don't want to send custom language files
+in a resource pack, server-side translations are the only alternative. They work anywhere `Component` API exists, except for ItemStack display text like the display name or lore. So
+chat, entity display names, scoreboards, tab lists, etc. all support translations.
+
+:::caution
+The player's language as declared in the settings packet sent by the client arrives **after** the player has joined the server, so there are no guarentees that
```
Comment (kennytv) [inline docs/paper/dev/api/component-api/i18n.md:31]:
```suggestion
The player's language as declared in the settings packet sent by the client arrives **after** the player has joined the server, so there are no guarantees that
```

File: docs/paper/dev/api/component-api/i18n.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+All translation is done through the [GlobalTranslator](https://jd.advntr.dev/api/latest/net/kyori/adventure/translation/GlobalTranslator.html). You can render translations yourself and add new sources
+for translations.
+
+You can add sources to the `GlobalTranslator` by creating instances of [TranslationRegistry](https://jd.advntr.dev/api/latest/net/kyori/adventure/translation/TranslationRegistry.html) or implementing
+the [Translator](https://jd.advntr.dev/api/latest/net/kyori/adventure/translation/Translator.html) interface yourself.
+
+## Where translations work
+
+Vanilla minecraft handles translations on the client by using the language files bundled with the client or provided via resource packs. If you don't want to send custom language files
```
Comment (kennytv) [inline docs/paper/dev/api/component-api/i18n.md:26]:
```suggestion
Vanilla Minecraft handles translations on the client by using the language files bundled with the client or provided via resource packs. If you don't want to send custom language files
```

**Review by kennytv — COMMENTED** [review]
Also needs at least a short code example

**Comment by Machine-Maker** [comment]
Ok, added an example using `ResourceBundle`. Do we need to add an example of a custom `Translator` implementation? I don't really think its needed.

File: docs/paper/dev/api/component-api/i18n.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+
+## Examples
+
+### ResourceBundle
+```java
+TranslationRegistry registry = TranslationRegistry.create(Key.key("namespace:value"));
+
+ResourceBundle bundle = ResourceBundle.getBundle("class.path.lang.en_us", Locale.US, UTF8ResourceBundleControl.get());
+registry.registeryAll(Locale.US, bundle, true);
```
Comment (olijeffers0n) [inline docs/paper/dev/api/component-api/i18n.md:44]: `registeryAll` -> `registerAll`

**Comment by olijeffers0n** [comment]
As someone who hasn't used translatable components before, I would want an example of how to send these / create these components. Also, an explanation to what a translation key and/or what the translation files look like.

File: docs/paper/dev/api/component-api/i18n.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+for translations.
+
+You can add sources to the `GlobalTranslator` by creating instances of [TranslationRegistry](https://jd.advntr.dev/api/latest/net/kyori/adventure/translation/TranslationRegistry.html) or implementing
+the [Translator](https://jd.advntr.dev/api/latest/net/kyori/adventure/translation/Translator.html) interface yourself.
+
+## Where translations work
+
+Vanilla Minecraft handles translations on the client by using the language files bundled with the client or provided via resource packs. If you don't want to send custom language files
+in a resource pack, server-side translations are the only alternative. They work anywhere `Component` API exists, except for ItemStack display text like the display name or lore. So
+chat, entity display names, scoreboards, tab lists, etc. all support translations.
```
[the remark this answers is not in the document]
Comment (xpple) [reply docs/paper/dev/api/component-api/i18n.md:28]: It is correct English.

[the remark this answers is not in the document]
Comment (Lulu13022002) [reply docs/paper/dev/api/component-api/i18n.md:28]: Indeed

[the remark this answers is not in the document]
Comment (Machine-Maker) [reply docs/paper/dev/api/component-api/i18n.md:28]: Actually, I think its supposed to have a comma after it if its used in the middle of a sentence. So I'll change it to `...tab lists, etc., all support...`

**Comment by Machine-Maker** [comment]
> As someone who hasn't used translatable components before, I would want an example of how to send these / create these components. Also, an explanation to what a translation key and/or what the translation files look like.

Good call. We can talk about the `Translatable` interface here, and all the api types that implement that.

As for an example of a resource bundle file... that is just one implementation. You can create translations from any format you want, its not closed off.

**Review by olijeffers0n — CHANGES_REQUESTED** [review]
Looks good, the only thing I would say is that it probably needs an example of using a translatable component

**Review by olijeffers0n — APPROVED** [review]
Looks good to me

**Pull request merged**